### ISSUE TITLE
[9.1 beta] Demangling VS C++ functions not working in autoanalysis

### ISSUE THREAD

**Issue by Piruzzolo** [NationalSecurityAgency/ghidra#1057]
**Describe the bug**
After autoanalisys of a .exe win executable (demangle enabled, importing PDB with debug symbols), one can see that all the mangled C++ symbols are not demangled. The exe was compiled with Visual Studio.
I've tried to run the demangler separately after, but no luck - it doesn't work and does show this message:  

```
Demangle Symbol> Unable to demangle symbol at 007caf60; name: ??_C@_0BJ@JOACEOKL@Fri?5Dec?5?57?516?314?334?52012?$AA@.  Message: Symbol name matches possible default symbol name: s_Fri_Dec__7_16:14:34_2012
Demangle Symbol> Unable to demangle symbol at 007db9a4; name: ??_C@_04KIACLCF@1280?$AA@.  Message: Symbol name matches possible default symbol name: s_1280
Demangle Symbol> Unable to demangle symbol at 007db9ac; name: ??_C@_04MJHCKDHM@1024?$AA@.  Message: Symbol name matches possible default symbol name: s_1024
Demangle Symbol> Unable to demangle symbol at 007e9454; name: ??_C@_04GGHKDNGG@4000?$AA@.  Message: Symbol name matches possible default symbol name: s_4000
Demangle Symbol> Unable to demangle symbol at 007e945c; name: ??_C@_04FLBKBENG@5000?$AA@.  Message: Symbol name matches possible default symbol name: s_5000
Demangle Symbol> Unable to demangle symbol at 007e9464; name: ??_C@_04BMLKGOAG@6000?$AA@.  Message: Symbol name matches possible default symbol name: s_6000
Demangle Symbol> Unable to demangle symbol at 007e946c; name: ??_C@_04CBNKEHLG@7000?$AA@.  Message: Symbol name matches possible default symbol name: s_7000
Demangle Symbol> Unable to demangle symbol at 007e9474; name: ??_C@_04KDIKNAGH@8000?$AA@.  Message: Symbol name matches possible default symbol name: s_8000
Demangle Symbol> Unable to demangle symbol at 007e947c; name: ??_C@_04JOOKPJNH@9000?$AA@.  Message: Symbol name matches possible default symbol name: s_9000
Demangle Symbol> Unable to demangle symbol at 007e9484; name: ??_C@_05CLDMBJBA@10000?$AA@.  Message: Symbol name matches possible default symbol name: s_10000
Demangle Symbol> Unable to demangle symbol at 007e948c; name: ??_C@_05BGFMDAKA@11000?$AA@.  Message: Symbol name matches possible default symbol name: s_11000
Demangle Symbol> Unable to demangle symbol at 007e9494; name: ??_C@_05FBPMEKHA@12000?$AA@.  Message: Symbol name matches possible default symbol name: s_12000
Demangle Symbol> Unable to demangle symbol at 007e949c; name: ??_C@_05GMJMGDMA@13000?$AA@.  Message: Symbol name matches possible default symbol name: s_13000
Demangle Symbol> Unable to demangle symbol at 007e94a4; name: ??_C@_05NOLMLPNA@14000?$AA@.  Message: Symbol name matches possible default symbol name: s_14000
Demangle Symbol> Unable to demangle symbol at 007fd4f0; name: ??_C@_04ECAGGAOJ@defb?$AA@.  Message: Symbol name matches possible default symbol name: s_defb

```
Params are named correctly, but their type is not right - should be inferred from the signature I guess.

So, I've separately run the `DemangleAllScript.java` script and it **worked**. Same error messages as above.

**Expected behavior**
Function names demangled during autoanalysis.

**Environment (please complete the following information):**
 - OS: Windows 10 latest update
 - Java Version: 11
 - Ghidra Version: 9.1 beta - 24/09

**Comment by ghizard** [comment]
This may not be what I thought I saw before.  The issue above (```Symbol name matches possible default symbol name```) is specific to ReservedDynamicLabelName in Ghidra, which I believe needs internal Ghidra team discussion before determining what to do about it.

**Comment by Piruzzolo** [comment]
_**Update**_: still present in 9.1 release. 
@ghizard any news?

**Comment by ryanmkurtz** [comment]
Just talked to @ghizard.  The issue is assigned internally, but not yet resolved.

**Comment by aboood40091** [comment]
Any update on this?

> This may not be what I thought I saw before. The issue above (`Symbol name matches possible default symbol name`) is specific to ReservedDynamicLabelName in Ghidra, which I believe needs internal Ghidra team discussion before determining what to do about it.

**Comment by ghizard** [comment]
Checked with person who would do the work and he will try to look into the issue next week.  It is not really a demangler issue, but an issue that stems from how certain symbols are not allowed to be laid down because they fit the pattern of some reserved symbols.  I'd like to believe that we can remove the restriction, as I "think" we no longer write the reserved symbol pattern anymore.  Before duplicate symbols were allowed in Ghidra, Ghidra would give the name and tack on the address suffix.  All of the examples above look like that pattern.  s_deadbeef would get rejected because deadbeef looks like a hex address.

**Comment by aboood40091** [comment]
While the issue does seem to stem from reserved symbols, I don't think that's the exact format as this one fails as well: `Symbol name matches possible default symbol name: s_Fri_Dec__7_16:14:34_2012`

Anyway, my humble opinion is that this restriction should no longer exist as Ghidra now allows for duplicate symbols.

**Comment by ghizard** [comment]
This got fixed in 9.2, but the ticket never got closed out properly.  The release note will show up in Ghidra 9.2.1 under GT-3185.
